freetextresponse/freetextresponse.py

```diff
@@ -459,45 +500,81 @@ def _get_submit_class(self):
             result = 'nodisplay'
         return result
 
-    def _get_submitted_message(self):
+    def _get_submitted_message(self, submit_type=None):
         """
         Returns the message to display in the submission-received div
         """
         result = ''
-        if self.count_attempts > 0 and self._word_count_valid():
+        if 'submit' == submit_type and self._word_count_valid():
             result = self.submitted_message
         return result
 
+    def _get_user_alert(self, submit_type=None, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert(TBD) div
+        """
+        result = ''
+        if 'submit' == submit_type and not self._word_count_valid():
+            result = self._get_word_count_message(ignore_attempts)
+        elif 'save' == submit_type:
+            result = self.saved_message
+        return result
+
     @XBlock.json_handler
     def submit(self, data, suffix=''):
         # pylint: disable=unused-argument
         """
         Processes the user's submission
         """
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
-            raise StandardError(
-                _(
-                    'User has already exceeded the '
-                    'maximum number of allowed attempts'
-                )
-            )
-        self.student_answer = data['student_answer']
-        if self._word_count_valid():
-            if self.max_attempts == 0:
-                self.count_attempts = 1
-            else:
-                self.count_attempts += 1
-            self._compute_score()
+        if ((0 == self.max_attempts) or
+                (0 < self.max_attempts and
+                    self.count_attempts < self.max_attempts)):
+            self.student_answer = data['student_answer']
+            submit_type = 'submit'
+            if self._word_count_valid():
+                if self.max_attempts == 0:
+                    self.count_attempts = 1
```

Comment:
I know this was already here, so just thinking out loud:
Why aren't we always counting attempts?
If `max_attempts` is unlimited (0), shouldn't we still keep count?
Why hard-code this to 1?